docs/development/plugins/data/public/kibana-plugin-plugins-data-public.querystringinput.md

```diff
@@ -7,5 +7,5 @@
 <b>Signature:</b>
 
 ```typescript
-QueryStringInput: React.FC<Pick<Props, "placeholder" | "onBlur" | "onChange" | "onSubmit" | "query" | "size" | "indexPatterns" | "prepend" | "screenTitle" | "dataTestSubj" | "disableAutoFocus" | "persistedLog" | "bubbleSubmitEvent" | "languageSwitcherPopoverAnchorPosition" | "onChangeQueryInputFocus">>
+QueryStringInput: React.FC<Pick<Props, "query" | "prepend" | "size" | "className" | "placeholder" | "onChange" | "onBlur" | "onSubmit" | "indexPatterns" | "dataTestSubj" | "screenTitle" | "disableAutoFocus" | "persistedLog" | "bubbleSubmitEvent" | "languageSwitcherPopoverAnchorPosition" | "onChangeQueryInputFocus">>
```

Comment:
do you have the same problem with latest master? https://github.com/elastic/kibana/pull/76890

Reply:
note that this type doesn't provide a lot of value and it's better to declare it explicitly https://github.com/microsoft/rushstack/issues/1958#issuecomment-653246015

Reply:
I changed `export const QueryStringInput = withKibana(QueryStringInputUI);` to `export const QueryStringInput = withKibana<Props>(QueryStringInputUI);` with no change in the resulting docs.

Reply:
Right, because `withKibana` is inferring the result type with `Omit`.
Before:
```typescript
const QueryStringInput: React.FC<Pick<Props, "query" | "prepend" ....>
```
After:
```typescript
const QueryStringInput: React.FC<Pick<Props, "query" | "prepend" ....>
```
The main point that the generated document doesn't provide any useful information about the inferred interface. It's hard to say what is a type of `query` property, for example.
For comparison: the output with an explicit `QueryStringInputProps` interface https://github.com/elastic/kibana/pull/77054

Reply:
Oh, right right.